### DISCUSSION TITLE
Publish wiki directly from repo code

### DISCUSSION THREAD

**Discussion by bisdevres** [community/community#27486] [Other Feature Feedback, Questions, & Ideas]
Just like in Azure DevOps, it must be possible to publish the wiki directly from the repo source code.
👉 [Publish a Git repository to a wiki](https://docs.microsoft.com/en-us/azure/devops/project/wiki/publish-repo-to-wiki?view=azure-devops&tabs=browser#publish-a-git-repository-to-a-wiki-1)

please implement this feature like is done in Azure DevOps

**Comment by MathiasReker** [comment]
Isn't this possible using GitHub actions?

**Reply by bisdevres** [reply]
yes it is.

but somehow i'm still hoping they would do a similar setup to GitHub pages.
There you select a branch / folder that should be used for creating the pages.

i feel like GitHub wiki should work with a similar setup.

**Reply by bisdevres** [reply]
and currently there can be merge conflicts when the code repo (not wiki) is edited while the wiki is also edited.

all these problems are avoided when working with a setup like in azure devops.

**Comment by airtower-luna** [comment]
Wiki pages live in a Git repository, which you can [fetch and push](https://docs.github.com/en/communities/documenting-your-project-with-wikis/adding-or-editing-wiki-pages#adding-or-editing-wiki-pages-locally). I just did a bit of experimenting, and you can perfectly well use `actions/checkout` on that repository, edit, commit and push. Here's what I did:

https://github.com/airtower-luna/hello-github-actions/blob/4b0e7667be58a0b4a31c1624b7ac1f5a0e258985/.github/workflows/wiki-bot.yaml

You could copy files over etc. instead of running the bot script. And of course adjust the files to add.
